1. Print out a data type of a variable?

In [2]:
st_var = "Hello World"
print(type(st_var))

b_var = False
print(type(b_var))

<class 'str'>
<class 'bool'>


2. Round 4.5667 to the nearest hundreth using a build-in function, round().

In [3]:
num = 4.5667
print(round(num, 2))

4.57


3. Convert "657" to an integer data type

In [5]:
int("657")

657

4. Write a function to calculate the miles per gallon.
Get a miles driven and gallons used from a user and call your function to calculate the miles per gallon. 

In [9]:
def miles_per_gallon():
    print("Enter miles driven:")
    miles = float(input())
    print("Enter gallons of gas used:")
    gallons = float(input())
    mpg = miles / gallons
    print(f"Miles Per Gallon: {mpg}")
    
miles_per_gallon()

Enter miles driven:
150
Enter gallons of gas used:
7.5
Miles Per Gallon: 20.0


5. Enhance your function from exercise 3. Additionally, get a gas price per gallon from a user and write a program to compute the total gas cost and cost per mile. 

The output should look something like this:

    Enter miles driven:            150
    Enter gallons of gas used:     15
    Enter cost per gallon:          3

    Miles Per Gallon:   10.0
    Total Gas Cost:     45.0
    Cost per Mile:      0.3


In [16]:
def miles_per_gallon():
    
    # input all required data
    print("Enter miles driven:", end = '\t\t')
    miles = float(input())
    print("Enter gallons of gas used:", end = '\t')
    gallons = float(input())
    print("Enter cost per gallon:", end = '\t\t')
    cost_per_gallon = float(input())
    
    # make calculations
    mpg = miles / gallons
    gas_cost = round(gallons * cost_per_gallon, 2)
    cost_per_mile = round(gas_cost / miles, 2)
    
    #print results
    print(f"Miles Per Gallon:\t{mpg}")
    print(f"Total Gas Cost:\t\t{gas_cost}")
    print(f"Cost per Mile:\t\t{cost_per_mile}")
    
miles_per_gallon()

Enter miles driven:		120
Enter gallons of gas used:	4.8
Enter cost per gallon:		3.1
Miles Per Gallon:	25.0
Total Gas Cost:		14.88
Cost per Mile:		0.12


6. Write a function called multiply_two() that accepts two parameters. The function should print a message like '15 + 2 = 17'.  Randomly generate two numbers between 1 and 100, then call the function with the two random numbers.

In [30]:
import random as rand

def multiply_two(num1, num2):
    total_num = num1 + num2
    print (f"{num1} + {num2} = {total_num}")
    
number1 = rand.randint(1, 100)
number2 = rand.randint(1, 100)
multiply_two(number1, number2)
    

41 + 10 = 51


7. Write a function called make_shirt() that accepts a size and the text of a message. The function should print a sentence summarizing the size of the shirt and the message printed on it. The default shirt size is large and default message is 'I love Python!' Make a default size shirt with a message, a medium shirt with the default message, and a shirt of any size with a different message.

In [36]:
def make_shirt(size = 'large', message = 'I love Python!'):
    print(f"{message} in {size} size shirt")
    
make_shirt(message = "I feel good")
make_shirt(size = 'medium')
make_shirt(size = 'any', message = "I don't like rainy weather")


I feel good in large size shirt
I love Python! in medium size shirt
I don't like rainy weather in any size shirt


8. Create a function for a multiplication table.  Get a number from a user and then call you function which produces a multiplication table for that number.
The output should look something like this:

        Enter a number: 3
        3 X 1 = 3
        3 X 2 = 6
        3 X 3 = 9
        3 X 4 = 12
        3 X 5 = 15
        3 X 6 = 18
        3 X 7 = 21
        3 X 8 = 24
        3 X 9 = 27
        

In [42]:
def multiplication_table(num):
    print(f"{num} X 1 = {num * 1}")
    print(f"{num} X 2 = {num * 2}")
    print(f"{num} X 3 = {num * 3}")
    print(f"{num} X 4 = {num * 4}")
    print(f"{num} X 5 = {num * 5}")
    print(f"{num} X 6 = {num * 6}")
    print(f"{num} X 7 = {num * 7}")
    print(f"{num} X 8 = {num * 8}")
    print(f"{num} X 9 = {num * 9}")

print("Enter a number:", end = ' ')
number = int(input())
multiplication_table(number)

Enter a number: 7
7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63


9. Write a function that accepts a string and returns a string that is 
    - without any leading nor trailing blanks
    - converts to all capital letters
    - and replaces all blanks with a period(.).  

Write a program to accept a string from a user then call your function and display the original string and changed string.  

The output should look something like this:

        Enter a text: may your dreams come true
        
        Original Text: may your dreams come true
        New Text: MAY.YOUR.DREAMS.COME.TRUE

In [10]:
def str_custom_convert(strline):
    return strline.strip().upper().replace(' ', '.')
    
print('Enter a text: ', end = '')
org_text = input()
new_text = str_custom_convert(org_text)
print('\nOriginal Text: ' + org_text)
print('New Text: ' + new_text)

Enter a text:    what you see is what you get    

Original Text:    what you see is what you get    
New Text: WHAT.YOU.SEE.IS.WHAT.YOU.GET


10. Write a function that returns a randomly selected character from a passed string. Write a program to accept a string from a user then call your function and display the original string and selected character.

The output should look something like this:

        Enter a text: functions
        Original Text:	functions
        Random character:	s

In [21]:
import random as rand

def get_character(str):
    strlen = len(str)
    char_position = int(rand.randint(1, strlen))
    return str[char_position - 1]

print('Enter a text: ', end = '')
org_text = input()
character = get_character(org_text)
print("Original Text: ", org_text)
print("Random Character: ", character)

Enter a text: Computer
Original Text:  Computer
Random Character:  u
